File: omc/android/app/src/main/java/com/pureenee/model/ProdottoDeal.java
package com.pureenee.model;

import org.json.JSONException;
import org.json.JSONObject;

import java.util.List;

import io.objectbox.annotation.Backlink;
import io.objectbox.annotation.Entity;
import io.objectbox.annotation.Id;
import io.objectbox.relation.ToMany;
import io.objectbox.relation.ToOne;

@Entity
public class ProdottoDeal {

    @Id
    private long id;

    ToOne<Prodotto> prodotto;

    ToOne<Deal> deal;

    @Backlink(to = "prodottoDeal")
    ToMany<ProdottoOrdine> prodottiOrdine;

    private double quantitativoProdottoDeal;

    private double disponibilitaPersonale;

    private double disponibilitaMercato;

    private double quantitativoProdottoDealIniziale;

    private double disponibilitaMercatoIniziale;

    private double disponibilitaPersonaleIniziale;

    private double prezzoIngrosso;

    private double prezzoDettaglio;

    private double importoInvestito;

    private double importoRientrato;

    public ProdottoDeal(

    ) {
    }

    public Prodotto getProdotto(){
        return this.prodotto.getTarget();
    }

    public void setProdotto(Prodotto prodotto){
        this.prodotto.setTarget(prodotto);
    }

    public Deal getDeal(){
        return this.deal.getTarget();
    }

    public void setDeal(Deal deal){
        this.deal.setTarget(deal);
    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public double getPrezzoIngrosso() {return prezzoIngrosso;}

    public void setPrezzoIngrosso(double prezzoIngrosso) {this.prezzoIngrosso = prezzoIngrosso;}

    public double getPrezzoDettaglio() {return prezzoDettaglio;}

    public void setPrezzoDettaglio(double prezzoDettaglio) {
        this.prezzoDettaglio = prezzoDettaglio;
    }

    public double getQuantitativoProdottoDeal() {
        return quantitativoProdottoDeal;
    }

    public void setQuantitativoProdottoDeal(double quantitativoProdottoDeal) {
        this.quantitativoProdottoDeal = quantitativoProdottoDeal;
    }

    public double getDisponibilitaPersonale() {
        return disponibilitaPersonale;
    }

    public void setDisponibilitaPersonale(double disponibilitaPersonale) {
        this.disponibilitaPersonale = disponibilitaPersonale;
    }

    public double getDisponibilitaMercato() {
        return disponibilitaMercato;
    }

    public void setDisponibilitaMercato(double disponibilitaMercato) {
        this.disponibilitaMercato = disponibilitaMercato;
    }

    public double getQuantitativoProdottoDealIniziale() {
        return quantitativoProdottoDealIniziale;
    }

    public void setQuantitativoProdottoDealIniziale(double quantitativoProdottoDealIniziale) {
        this.quantitativoProdottoDealIniziale = quantitativoProdottoDealIniziale;
    }

    public double getDisponibilitaMercatoIniziale() {
        return disponibilitaMercatoIniziale;
    }

    public void setDisponibilitaMercatoIniziale(double disponibilitaMercatoIniziale) {
        this.disponibilitaMercatoIniziale = disponibilitaMercatoIniziale;
    }

    public double getDisponibilitaPersonaleIniziale() {
        return disponibilitaPersonaleIniziale;
    }

    public void setDisponibilitaPersonaleIniziale(double disponibilitaPersonaleIniziale) {
        this.disponibilitaPersonaleIniziale = disponibilitaPersonaleIniziale;
    }

    public double getImportoInvestito() {
        return importoInvestito;
    }

    public void setImportoInvestito(double importoInvestito) {
        this.importoInvestito = importoInvestito;
    }

    public double getImportoRientrato() {
        return importoRientrato;
    }

    public void setImportoRientrato(double importoRientrato) {
        this.importoRientrato = importoRientrato;
    }

    public List<ProdottoOrdine> getProdottiOrdine(){
        return this.prodottiOrdine;
    }

}


//
    //private double prezzoIngrosso;
//
    //private double prezzoDettaglio;
//
    //private double importoInvestito;
//
    //private double importoRientrato;

File: omc/lib/model/prodotto_deal.dart
import 'package:omc/model/prodotto.dart';


class ProdottoDeal {
  final int id;
  final Prodotto prodotto;
  final double quantitativoProdottoDeal;
  final double disponibilitaPersonale;
  final double disponibilitaMercato;
  final double quantitativoProdottoDealIniziale;
  final double disponibilitaMercatoIniziale;
  final double disponibilitaPersonaleIniziale;
  final double prezzoIngrosso;
  final double prezzoDettaglio;
  final double importoInvestito;
  final double importoRientrato;


  ProdottoDeal(
      this.id,
      this.prodotto,
      this.quantitativoProdottoDeal,
      this.disponibilitaPersonale,
      this.disponibilitaMercato,
      this.quantitativoProdottoDealIniziale,
      this.disponibilitaMercatoIniziale,
      this.disponibilitaPersonaleIniziale,
      this.prezzoIngrosso,
      this.prezzoDettaglio,
      this.importoInvestito,
      this.importoRientrato
      );

  factory ProdottoDeal.fromJson(Map<String, dynamic> json) {
    return ProdottoDeal(
      json['id'],
      Prodotto.fromJson(json['prodotto']),
      json['quantitativoProdottoDeal'],
      json['disponibilitaPersonale'],
      json['disponibilitaMercato'],
      json['quantitativoProdottoDealIniziale'],
      json['disponibilitaMercatoIniziale'],
      json['disponibilitaPersonaleIniziale'],
      json['prezzoIngrosso'],
      json['prezzoDettaglio'],
      json['importoInvestito'],
      json['importoRientrato'],
    );

  }
  Map<String, dynamic> toJson() => {
    'id': id,
    'prodotto': prodotto.toJson(),
    'quantitativoProdottoDeal':quantitativoProdottoDeal,
    'disponibilitaPersonale': disponibilitaPersonale,
    'disponibilitaMercato':disponibilitaMercato,
    'quantitativoProdottoDealIniziale':quantitativoProdottoDealIniziale,
    'disponibilitaMercatoIniziale':disponibilitaMercatoIniziale,
    'disponibilitaPersonaleIniziale':disponibilitaPersonaleIniziale,
    'prezzoIngrosso':prezzoIngrosso,
    'prezzoDettaglio': prezzoDettaglio,
    'importoInvestito': importoInvestito,
    'importoRientrato': importoRientrato,
  };

  bool isEsaurito(){
    return disponibilitaMercato <= 0;
  }
}